systemtest/huge_traces_test.go

```diff
@@ -0,0 +1,54 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package systemtest_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/apm-server/systemtest"
+	"github.com/elastic/apm-server/systemtest/apmservertest"
+	"github.com/elastic/apm-server/systemtest/estest"
+)
+
+func TestTransactionDroppedSpansStats(t *testing.T) {
+	systemtest.CleanupElasticsearch(t)
+	srv := apmservertest.NewUnstartedServer(t)
+	srv.Config.Aggregation = &apmservertest.AggregationConfig{
+		ServiceDestinations: &apmservertest.ServiceDestinationAggregationConfig{
+			Enabled:  true,
+			Interval: time.Second,
+		},
+	}
+	err := srv.Start()
+	require.NoError(t, err)
+
+	systemtest.SendBackendEventsPayload(t, srv, "../testdata/intake-v2/transactions.ndjson")
```

Comment:
It would be great if we could isolate only the relevant transactions/spans. Perhaps eventually we should use the Go agent directly, if/when we've implemented support for composite and dropped spans stats there.

Reply:
I was unsure whether to copy the transaction that I added in the file or use the whole file, it does create unnecessary data in Elasticsearch, I could do that and add a comment to switch to the Agent implementation when available.

We could also add a filter on fields to these ingesting functions so that we can filter on the test. I can change the signature to accept a variadic string argument which can be used to filter on "<key>=<val>".

Reply:
How about we introduce a new testdata file with just one transaction with dropped spans for now (then no need to filter, and some other changes to tests can be reverted I think), and a TODO to switch over to the Go agent when available?

Reply:
sounds good, I'll do that then.